src/CoreWCF.WebHttp/src/CoreWCF/Configuration/WebHttpTransportServiceBuilder.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using CoreWCF.OpenApi;
+using CoreWCF.OpenApi.Middleware;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+
+namespace CoreWCF.Configuration
+{
+    internal class WebHttpTransportServiceBuilder : ITransportServiceBuilder
+    {
+        private bool _configured = false;
+        private readonly object _lock = new object();
+
+        public void Configure(IApplicationBuilder app)
+        {
+            if (!_configured)
+            {
+                lock (_lock)
+                {
+                    if (!_configured)
+                    {
+                        ConfigureCore(app);
+                        _configured = true;
+                    }
+                }
+            }
+        }
+
+        private void ConfigureCore(IApplicationBuilder app)
+        {
+            OpenApiDocumentProvider documentProvider = app.ApplicationServices.GetRequiredService<OpenApiDocumentProvider>();
+            if (documentProvider.Contracts.Count > 0)
+            {
+                app
+                    .UseStaticFiles(new StaticFileOptions
+                    {
+                        FileProvider = new ManifestEmbeddedFileProvider(typeof(OpenApiOptions).Assembly, @"CoreWCF\OpenApi\swagger-ui"),
+                        RequestPath = "/swagger"
```

Comment:
I spoke with someone on the aspnet core team and while they didn't know whether this is a good idea or not, I believe it isn't. This class was created as a file provider to loading configuration, the intended purpose wasn't for serving files. I had a look at the code and it reloads a stream from the manifest every single time. I think this is going to be a performance problem. I've been looking into alternatives and there's capability to embed content files into packages. You put them into a folder called wwwroot in the package and it appears at runtime under _content/packageid. There's some helper code for this, but it wasn't created until .NET Core 3.0. You don't need to reference any api's to use it, just place the files in the wwwroot folder in the generated package. This was designed for use with razor pages so I'm not sure if there's more to it. I'll keep digging to find more info, but I think this way of serving files isn't the correct way to go.

Reply:
I understand your concerns. I switched this to a model that reads the manifest resources once and serves them from memory after that. It was necessary to fit in with the DI pipeline anyways.